src/http/websocket_http_client.zig

```diff
@@ -1419,16 +1431,20 @@ pub fn NewWebSocketClient(comptime ssl: bool) type {
             var mask_buf: *[4]u8 = final_body_bytes[2..6];
             std.mem.writeIntSliceBig(u16, final_body_bytes[6..8], code);
 
+            var reason = JSC.ZigString.Empty;
             if (body) |data| {
-                if (body_len > 0) @memcpy(final_body_bytes[8..][0..body_len], data[0..body_len]);
+                if (body_len > 0) {
+                    reason = JSC.ZigString.init(data[0..body_len]);
```

Comment:
I don't think this is correct? It should use the copy from final_body_bytes instead of data. data is temporary. This will segfault if it's not cloned.